File: DemoMethods/PythonDemo/demo_methods/Basic/BoostingEnabled.py
from demo.holder import DocumentStoreHolder
from demo.entities import *


class BoostingEnabled(object):
    @staticmethod
    def boosting_enabled(city="London", country="Denmark"):
        with DocumentStoreHolder.get_store().open_session() as session:
            orders = list(session.query(object_type=Order, index_name="OrderByCompanyAndCountryWithBoost",
                                        wait_for_non_stale_results=True,
                                        nested_object_types={"OrderedAt": datetime, "RequireAt": datetime,
                                                             "ShippedAt": datetime, "ShipTo": AddressC,
                                                             "Lines": OrderLine}).
                          where(ShipTo_City=city, ShipTo_Country=country))
						  
            return orders

File: DemoServer/Demos/Basic/BoostingEnabled.cs
using System.Linq;
using DemoServer.Controllers;
using DemoServer.Entities;
using DemoServer.Helpers;
using DemoServer.Indexes;
using Microsoft.AspNetCore.Mvc;

namespace DemoServer.Demos.Basic
{
    public partial class BasicController : BaseController
    {
        [HttpGet]
        [Route("/basic/boostingEnabled")]
        [Demo("Boosting Enabled", DemoOutputType.Flatten, demoOrder: 140)]
        public object BoostingEnabled(string city = "London", string country = "Denmark")
        {
            using (var session = DocumentStoreHolder.Store.OpenSession())
            {
                var orders = session.Query<Order, OrderByCompanyAndCountryWithBoost>()
                    .Where(x => x.ShipTo.City == city || x.ShipTo.Country == country)
                    .ToList();

                return orders;
            }
        }
    }
}
